Sources/LiveKit/Track/Recorders/LocalAudioTrackRecorder.swift

```
@@ -0,0 +1,138 @@
+@objc
+public final class LocalAudioTrackRecorder: NSObject, AudioRenderer {
+    public typealias Stream = AsyncStream<Data>
+
+    /// The local audio track to capture audio from.
+    @objc
+    public let track: LocalAudioTrack
+
+    /// The format of the audio data to stream.
+    @objc
+    public let format: AVAudioCommonFormat
+
+    /// The sample rate of the audio data to stream.
+    @objc
+    public let sampleRate: Int
+
+    /// The number of channels of the audio data to stream.
+    @objc
+    public let channels: Int = 1
+
+    /// The maximum size of the audio data to buffer.
+    @objc
+    public let maxSize: Int
+
+    private let _state = StateSync<State>(State())
+    private struct State {
+        var continuation: Stream.Continuation?
+    }
+
+    /// Initialize the audio recorder with a local audio track.
+    /// - Parameters:
+    ///   - track: The local audio track to capture audio from.
+    ///   - format: The format of the audio data to stream.
+    ///   - sampleRate: The sample rate of the audio data to stream.
+    ///   - maxSize: The maximum size of the audio data to buffer.
+    /// - Note: The default maximum size is 0, which means that the audio data will be buffered indefinitely.
+    @objc
+    public init(track: LocalAudioTrack, format: AVAudioCommonFormat, sampleRate: Int, maxSize: Int = 0) {
+        self.track = track
+        self.format = format
+        self.sampleRate = sampleRate
+        self.maxSize = maxSize
+    }
+
+    /// Starts capturing audio from the local track and returns a stream of audio data.
+    /// - Returns: A stream of audio data.
+    /// - Throws: An error if the audio track cannot be started.
+    public func start() async throws -> Stream {
+        guard _state.continuation == nil else {
+            throw LiveKitError(.invalidState, message: "Cannot start the recorder multiple times.")
+        }
+
+        try await track.startCapture()
```

Forking that into a new task would give us some marginal speedup, for the sake of discarding errors (and converting it to `Stream?` as a did before).

---

I think this makes sense, but I would recommend logging any errors thrown by `startCapture`.